src/SignalR/common/Protocols.MessagePack/src/MessagePackHubProtocolOptions.cs

```diff
@@ -9,24 +9,24 @@ namespace Microsoft.AspNetCore.SignalR
 {
     public class MessagePackHubProtocolOptions
     {
-        private IList<IFormatterResolver> _formatterResolvers;
+        private MessagePackSerializerOptions _messagePackSerializerOptions;
 
-        public IList<IFormatterResolver> FormatterResolvers
```

Comment:
Can we keep FormatterResolvers around? Maybe deprecate it?

Reply:
Why keep it around? It wouldn't avoid a binary breaking change because the MessagePack assembly itself that defines the `IFormatterResolver` type has lots of binary breaking changes in v2.x.
If we *did* keep it around, we'd have to define how to reconcile it with the `MessagePackSerializationOptions` that we also expose, and that options object is supposed to fully describe the *one* top-level resolver to be used.

Reply:
We would keep it around because removing it would require current customers to change their code when upgrading.

We normally deprecate old APIs for a major release, and then consider removing it in the next major release.

Reply:
TLDR : What do i do ? (as .Net5 is a breaking changed that's why i removed it, can add it back)

As i understand the general idea to deprecate instead of breaking, this specific scenario will make it look really weird and the reconciliation will add confusion and complexity to the code itself while you now have a direct access to "real formatter", "no tricks"

The  thing we "lost" though is the possibility to "Add/Insert" `FormatterResolver` to the pre-existing default ones I think
The new code would be (see https://github.com/dotnet/aspnetcore/issues/18569#issuecomment-601919087):
```
services
  .AddSignalR()
  .AddMessagePackProtocol(options =>
{
  // combining Formatter (looks weird to my eyes)
  options.SerializerOptions = MessagePackSerializerOptions.Standard.WithResolver(CompositeResolver.Create(new CustomFormatter(), options.SerializerOptions.Resolver));
});
```

The "Reconciliation" of both is going to be confusing for the user, 
**The order is very important as the first resolver found will be used**

The implementation will bring back previous logic that were removed because a direct access would make it unecessary

Just tell me if i keep it or not, (i honestly think dropping it would be less confusing)
For example,
* what the default for the `Deprecated` props ?
* what the default resolver for the `MessagePackSerializerOptions` ?
* do i reconcile both ? even if they both contain same resolvers ? (especially because of the "order"), who wins ?

Reply:
also, 
Aspnetcore 5 will already contain the update from `MessagePack 1.x` to `MessagePack 2.1.x` as a breaking change ;)

Reply:
> Aspnetcore 5 will already contain the update from `MessagePack 1.x` to `MessagePack 2.1.x` as a breaking change ;)

This is true, but we still try to limit breaking changes *in general*. Each individual change adds more friction for the updating customer.

Having said that, I'm generally pretty open to bigger breaking changes in things like Startup code (which is where most options configuration goes). I'd probably be OK submitting a proposal to our API review group to remove the Formatter Resolvers entirely, if we have a clear reasoning as to why (because the new MessagePack major version we use has richer configuration).

If it brings more clarity to customers to take the breaking change, that's a big point in favor.

Reply:
I agree keeping the old API will be more confusing than anything. And anyone who is customizing resolvers will almost certainly have other code changes to make because the resolvers had breaking changes in message pack 2.0.

Reply:
Let's remove it in this PR then and I'll raise it when we do the API review for this (which would be necessary either way ;)).

Reply:
regarding the API review, if it's decided to mark it as obsolete, i really suggest making an analyzer that will mark it as an error if both `MsgPackSerializerOptions` AND `List<FormaterResolver>` are used (no idea how thought ^^).

The same idea that was used in `aspnetcore3.0` when you tried to use `.UseMvc()` but did not disabled the Routing in `,AddMvc(options.EnabledRoutingBlabla = fakse)`

Reply:
Yep, there are some options there. We've done analyzers before, and we've also done things like logged warnings to the console. My gut feeling is that we're OK taking this breaking change, but we'll keep those options in mind.

Reply:
will read `https://github.com/dotnet/aspnetcore/blob/master/docs/APIReviewProcess.md` tomorrow and just wait for feedback IIUC ?